core/src/main/java/io/grpc/internal/RoundRobinServerList.java

```diff
@@ -81,32 +87,52 @@ T getTransportForNextServer() {
   }
 
   @VisibleForTesting
-  List<EquivalentAddressGroup> getList() {
+  public List<EquivalentAddressGroup> getList() {
     return list;
   }
 
-  int size() {
+  public int size() {
     return list.size();
   }
 
+  @Override
+  public boolean equals(Object other) {
+    if (!(other instanceof RoundRobinServerList)) {
+      return false;
+    }
+
+    @SuppressWarnings("unchecked")
+    RoundRobinServerList<T> otherCast = (RoundRobinServerList<T>) other;
+    return Objects.equal(tm, otherCast.tm) && Objects.equal(list, otherCast.list);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(tm, list);
+  }
+
   @NotThreadSafe
-  static class Builder<T> {
+  public static class Builder<T> {
     private final ImmutableList.Builder<EquivalentAddressGroup> listBuilder =
         ImmutableList.builder();
     private final TransportManager<T> tm;
 
-    Builder(TransportManager<T> tm) {
+    public Builder(TransportManager<T> tm) {
       this.tm = tm;
     }
 
     /**
      * Adds a server to the list, or {@code null} for a drop entry.
      */
-    void add(@Nullable InetSocketAddress addr) {
-      listBuilder.add(new EquivalentAddressGroup(addr));
+    public void add(@Nullable SocketAddress address) {
+      listBuilder.add(new EquivalentAddressGroup(address));
+    }
+
+    public void addAll(List<SocketAddress> addresses) {
```

Comment:
add() gets ambiguous with nulls, so I am going with addList, but happy to change again if that's not desirable either.